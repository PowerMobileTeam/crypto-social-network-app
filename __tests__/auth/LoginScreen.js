import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';

import LoginScreen from '../../src/auth/LoginScreen';

jest.mock('../../src/auth/AuthService');

jest.mock('../../src/auth/LoginForm', () => 'LoginForm');
jest.mock('../../src/auth/ForgotPassword', () => 'ForgotPassword');
jest.mock('../../src/common/components/VideoBackground', () => 'VideoBackground');

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('LoginScreen component', () => {

  it('should renders correctly', () => {
    const loginScreen = renderer.create(
      <LoginScreen />
    ).toJSON();
    expect(loginScreen).toMatchSnapshot();
  });

  it('should shows login form component', async () => {
    const wrapper = shallow(
      <LoginScreen />
    );

    // search login form
    let render = wrapper.dive();
    const loginForms = render.find('LoginForm');

    // should contain 1 login form
    expect(loginForms.length).toBe(1);
  });

  it('should shows forgot password component if the user press the button', async () => {

    const wrapper = shallow(
      <LoginScreen />
    );

    // search login form
    let render = wrapper.dive();
    const loginForms = render.find('LoginForm');

    // should contain 1 login form
    expect(loginForms.length).toBe(1);

    // simulate forgot password component button press
    loginForms.at(0).props().onForgot();

    // update component
    wrapper.update();
    render = wrapper.dive();

    const forgotPasswords = render.find('ForgotPassword');

    // should contain 1 forgot password
    expect(forgotPasswords.length).toBe(1);
  });
});
