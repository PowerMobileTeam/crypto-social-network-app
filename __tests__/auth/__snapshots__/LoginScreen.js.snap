// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen component should renders correctly 1`] = `
<View
  enabled={true}
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <VideoBackground />
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "padding": 10,
        },
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0.3,
            },
          ],
        }
      }
    >
      <Image
        collapsable={undefined}
        resizeMode="contain"
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "height": 100,
            "marginBottom": 30,
            "width": 200,
          }
        }
      />
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "translateY": 100,
            },
          ],
        }
      }
    >
      <LoginForm
        onForgot={[Function]}
        onLogin={[Function]}
        onRegister={[Function]}
      />
    </View>
  </View>
</View>
`;
